setup.py

```diff
@@ -1,15 +1,80 @@
 #!/usr/bin/env python
+import os
+import subprocess
+import sys
+from glob import glob
+from setuptools import setup, find_packages, Extension
 
-try:
-    from setuptools import setup
-except ImportError:
-    from distribute_setup import use_setuptools
-    use_setuptools()
-    from setuptools import setup
+
+if os.path.exists('relic'):
+    sys.path.insert(1, 'relic')
+    import relic.release
+else:
+    try:
+        import relic.release
+    except ImportError:
+        try:
+            subprocess.check_call(['git', 'clone',
+                'https://github.com/jhunkeler/relic.git'])
+            sys.path.insert(1, 'relic')
+            import relic.release
+        except subprocess.CalledProcessError as e:
+            print(e)
+            exit(1)
+
+
+version = relic.release.get_info()
+relic.release.write_template(version, 'lib/reftools')
 
 setup(
-    setup_requires=['d2to1>=0.2.9', 'stsci.distutils>=0.3.2'],
-    d2to1=True,
-    use_2to3=False,
-    zip_safe=False
+    name = 'reftools',
+    version = version.pep386,
```

Comment:
Post-commits are implicitly moving toward the next major/minor change.

1. Assume the repository was 1.7.dev50 (50 commit's ahead of the original 1.7 tag)
2. You tag the repository 2.0 because it's good enough to constitute a release (the code is now considered 2.0. Development continues toward 2.1 under the 2.0.devXX umbrella)